inspirehep/ext/cache_manager/cache.py

```diff
@@ -0,0 +1,146 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of INSPIRE.
+# Copyright (C) 2016 CERN.
+#
+# INSPIRE is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License as
+# published by the Free Software Foundation; either version 2 of the
+# License, or (at your option) any later version.
+#
+# INSPIRE is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with INSPIRE; if not, write to the Free Software Foundation, Inc.,
+# 59 Temple Place, Suite 330, Boston, MA 02111-1307, USA.
+
+import functools
+
+import logging
+
+from flask import current_app, request
+
+from flask_cache import Cache
+
+from invenio_base.globals import cfg
+
+logger = logging.getLogger(__name__)
+
+
+class InspireCache(Cache):
+    """Extends Cache from Flask Cache and adds mode-checking (debug or not) for memoize and cached methods.
+
+    The parameters in both inspire_memoize and inspire_cached are the same
+    as the parameters of the original methods in Flask Cache class
+
+    """
+
+    def inspire_memoize(self, timeout=None, make_name=None, unless=None):
+        def memoize(f):
+            @functools.wraps(f)
+            def decorated_function(*args, **kwargs):
+                if not cfg['DEBUG']:
```

Comment:
We checked with @jacquerie that `current_app.debug` has better performance so I'll replace `cfg["DEBUG"]` with it.